### ISSUE TITLE
Bug: sam build returns InvalidSamDocumentException for StateMachine-AutoPublishAlias

### ISSUE THREAD

**Issue by aidansteele** [aws/aws-sam-cli#5407]
### Description:
The serverless application model added support for the `AutoPublishAlias` on state machines [in v1.70.0](https://github.com/aws/serverless-application-model/releases/tag/v1.70.0). Using that new property causes SAM CLI v1.88.0 (latest at time of writing) to return an error during `sam build`.

### Steps to reproduce:
1. Create a template like the following:

```yaml
Transform: AWS::Serverless-2016-10-31

Resources:
  Example:
    Type: AWS::Serverless::StateMachine
    Properties:
      Type: STANDARD
      AutoPublishAlias: live # this is the problematic line
      Definition:
        StartAt: Start
        States:
          Start:
            Type: Pass
            End: true
```

2. Run `sam build`
3. Get an error

### Observed result:
```
aidan@Aidans-MacBook-Pro example % sam build --debug
2023-06-23 11:32:58,363 | Config file location: /Users/aidan/example/samconfig.toml                                                                  
2023-06-23 11:32:58,366 | Config file '/Users/aidan/example/samconfig.toml' does not exist                                                           
2023-06-23 11:32:58,367 | Using SAM Template at /Users/aidan/example/template.yml                                                                    
2023-06-23 11:32:58,385 | Using config file: samconfig.toml, config environment: default                                                                                 
2023-06-23 11:32:58,386 | Expand command line arguments to:                                                                                                              
2023-06-23 11:32:58,386 | --template_file=/Users/aidan/example/template.yml --mount_with=READ --build_dir=.aws-sam/build --cache_dir=.aws-sam/cache  
2023-06-23 11:32:58,455 | 'build' command is called                                                                                                                      
2023-06-23 11:32:58,457 | No Parameters detected in the template                                                                                                         
2023-06-23 11:32:58,470 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics                                     
2023-06-23 11:32:58,510 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': 'abc2a840-d17d-4e28-a625-db204ab48aca', 'installationId':                         
'70bca007-b2e7-41ee-8bbd-6a5df3cbd156', 'sessionId': 'b4a914eb-afcb-4c5e-95a8-60a7b82ee502', 'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.8.17',          
'samcliVersion': '1.88.0', 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam build', 'metricSpecificAttributes': {'projectType': 
'CFN', 'gitOrigin': None, 'projectName': '50d858e0985ecc7f60418aaf0cc5ab587f42c2570a884095a9e8ccacd0f6545c', 'initialCommit': None}, 'duration': 84, 'exitReason':       
'InvalidSamDocumentException', 'exitCode': 255}}]}                                                                                                                       
2023-06-23 11:32:59,243 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)               

Error: [InvalidResourceException('Example', 'property AutoPublishAlias not defined for resource of type AWS::Serverless::StateMachine')] ('Example', 'property AutoPublishAlias not defined for resource of type AWS::Serverless::StateMachine')
Traceback:
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/lib/telemetry/metric.py", line 184, in wrapped
    raise exception  # pylint: disable=raising-bad-type
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/lib/telemetry/metric.py", line 149, in wrapped
    return_value = func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/lib/utils/version_checker.py", line 42, in wrapped
    actual_result = func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/cli/main.py", line 95, in wrapper
    return func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/commands/build/command.py", line 179, in cli
    do_cli(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/commands/build/command.py", line 251, in do_cli
    with BuildContext(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/commands/build/build_context.py", line 184, in __enter__
    self.set_up()
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/commands/build/build_context.py", line 190, in set_up
    self._stacks, remote_stack_full_paths = SamLocalStackProvider.get_stacks(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/lib/providers/sam_stack_provider.py", line 255, in get_stacks
    current = SamLocalStackProvider(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/lib/providers/sam_stack_provider.py", line 51, in __init__
    self._template_dict = self.get_template(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/lib/providers/sam_base_provider.py", line 188, in get_template
    template_dict = SamTranslatorWrapper(template_dict, parameter_values=parameters_values).run_plugins()
  File "/opt/homebrew/Cellar/aws-sam-cli/1.88.0/libexec/lib/python3.8/site-packages/samcli/lib/samlib/wrapper.py", line 75, in run_plugins
    raise InvalidSamDocumentException(

An unexpected error was encountered while executing "sam build".
Search for an existing issue:
https://github.com/aws/aws-sam-cli/issues?q=is%3Aissue+is%3Aopen+Bug%3A%20sam%20build%20-%20InvalidSamDocumentException
Or create a bug report:
https://github.com/aws/aws-sam-cli/issues/new?template=Bug_report.md&title=Bug%3A%20sam%20build%20-%20InvalidSamDocumentException
aidan@Aidans-MacBook-Pro example % 
```

### Expected result:
That the SAM CLI will build, deploy etc successfully with a valid template.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)
```
{
  "version": "1.88.0",
  "system": {
    "python": "3.8.17",
    "os": "macOS-13.4.1-arm64-arm-64bit"
  },
  "additional_dependencies": {
    "docker_engine": "Not available",
    "aws_cdk": "Not available",
    "terraform": "1.5.1"
  },
  "available_beta_feature_env_vars": [
    "SAM_CLI_BETA_FEATURES",
    "SAM_CLI_BETA_BUILD_PERFORMANCE",
    "SAM_CLI_BETA_TERRAFORM_SUPPORT",
    "SAM_CLI_BETA_RUST_CARGO_LAMBDA"
  ]
}
```

**Comment by sriram-mv** [comment]
Thanks for the report! https://github.com/aws/serverless-application-model/releases/tag/v1.70.0 has been merged and is now in develop. This will out as part of the next release.

**Comment by benbpyle** [comment]
@aidansteele testing with sam-nightly and it's in there now :)

**Comment by mildaniel** [comment]
This issue should now be resolved in version 1.89.0. Give that a try and feel free to open a new issue if anything else comes up!

**Comment by adrianmazgaj** [comment]
Hey!
It seems like there are still some issues with "AWS::Serverless::StateMachine" resource. (SAM CLI, version 1.97.0) and codebuild with installed version: (SAM CLI 1.93.0). 
"property DeploymentPreference not defined for resource of type AWS::Serverless::StateMachine"
"property AutoPublishAlias not defined for resource of type AWS::Serverless::StateMachine"

Can You have a look?
Thanks!
